### PR TITLE
[PowerPC][AIX] Refactor existing logic to handle non-zero offsets for aix-small-local-dynamic-tls

### PR DESCRIPTION
**Pull request by orcguru**
To enable optimized small local-dynamic access sequence for non-zero offsets, this patch refactors existing 2a50921553798d2db52ca6330c89f0f8a5bc2215.

### CODE REVIEW

File: llvm/lib/Target/PowerPC/PPCISelDAGToDAG.cpp
Comment (orcguru) [inline llvm/lib/Target/PowerPC/PPCISelDAGToDAG.cpp:7631]: Similar check already exists in isEligible..., so I removed this one.

**Review by amy-kwan — APPROVED** [review]
The change and the removal of the section makes sense since we need to account for local-dynamic now, so LGTM.

**Comment by orcguru** [comment]
> The change and the removal of the section makes sense since we need to account for local-dynamic now, so LGTM.

Thank you Amy! I will wait for a few more days before merge.

**Pull request merged**
